crates/containerd-shim-wasm/src/container/context.rs

```diff
@@ -22,16 +18,22 @@ pub trait RuntimeContext {
     //   "/app/app.wasm#entry" -> { path: "/app/app.wasm", func: "entry" }
     //   "my_module.wat" -> { path: "my_module.wat", func: "_start" }
     //   "#init" -> { path: "", func: "init" }
-    fn wasi_entrypoint(&self) -> WasiEntrypoint;
+    fn entrypoint(&self) -> WasiEntrypoint;
 
-    fn wasm_layers(&self) -> &[WasmLayer];
+    fn wasi_loading_strategy(&self) -> WasiLoadingStrategy;
 
     fn platform(&self) -> &Platform;
 }
 
-pub struct WasiEntrypoint {
+pub enum WasiLoadingStrategy<'a> {
+    File(PathBuf),
+    Oci(&'a [WasmLayer]),
+}
+
+pub struct WasiEntrypoint<'a> {
     pub path: PathBuf,
     pub func: String,
+    pub arg0: Option<&'a Path>,
```

Comment:
Can you change the docs for `entrypoint()`?

Reply:
updated